src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,191 @@
+package frc.robot.subsystems.elevator;
+
+import java.util.TreeMap;
+
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkBase.SoftLimitDirection;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.SparkAbsoluteEncoder.Type;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+import frc.robot.utils.SendableHelper;
+import frc.robot.utils.SparkMaxUtils;
+
+//TODO absolute encoder zeroing
+
+public class Elevator extends SubsystemBase {
+  public enum ElevatorPosition{
+    HOME,
+    SCORE_TRAP,
+    SCORE_AMP,
+    PRE_CLIMB,
+    POST_CLIMB
+  }
+
+  public CANSparkMax leftMotor = new CANSparkMax(RobotMap.LEFT_ELEVATOR_CAN_ID, MotorType.kBrushless);
+  public CANSparkMax rightMotor = new CANSparkMax(RobotMap.RIGHT_ELEVATOR_CAN_ID, MotorType.kBrushless);
+
+  private final RelativeEncoder leftMotorEncoder = leftMotor.getEncoder();
+  private final AbsoluteEncoder leftMotorEncoderAbs = leftMotor.getAbsoluteEncoder(Type.kDutyCycle);
+  private final RelativeEncoder rightMotorEncoder = rightMotor.getEncoder();
+  private final AbsoluteEncoder rightMotorEncoderAbs = rightMotor.getAbsoluteEncoder(Type.kDutyCycle);
+
+  private ProfiledPIDController noteScoringElevatorController = new ProfiledPIDController(ElevatorCal.NOTE_SCORING_P,
+      ElevatorCal.NOTE_SCORING_I, ElevatorCal.NOTE_SCORING_D, new TrapezoidProfile.Constraints(
+          ElevatorCal.MAX_VELOCITY_IN_PER_SECOND,
+          ElevatorCal.MAX_ACCELERATION_IN_PER_SECOND_SQUARED));
+  private ProfiledPIDController chainGrabberElevatorController = new ProfiledPIDController(ElevatorCal.CLIMBING_P,
+      ElevatorCal.CLIMBING_I, ElevatorCal.CLIMBING_D, new TrapezoidProfile.Constraints(
+          ElevatorCal.MAX_VELOCITY_IN_PER_SECOND,
+          ElevatorCal.MAX_ACCELERATION_IN_PER_SECOND_SQUARED));
+
+  private ProfiledPIDController currentPIDController = noteScoringElevatorController;
+
+  private TreeMap<ElevatorPosition, Double> elevatorPositions;
+
+  private ElevatorPosition desiredPosition = ElevatorPosition.HOME;
+
+  private boolean usingNoteScoringPID = true;
+
+  private double prevVelocity = 0;
+  private double prevTimestamp = -1;
```

Comment:
This strategy is called using a "sentinel" value and it's generally less good than using a java optional for the empty state